front/src/AuthenticationScreen/index.js

```diff
@@ -0,0 +1,56 @@
+import React, { Component } from "react";
+import styled from "styled-components";
+import PropTypes from "prop-types";
+
+/**
+ * An authentication screen. Upon a submit it will send the token in the inputfield
+ * along with a function to handle the response message. Authentication screen is
+ * responsible for its unrendering.
+ * Author: S.A. Tanja
+ *
+ * Props:
+ *  - handleSubmit: A functional component that will send the token to the SRRS
+ */
+class AuthenticationScreen extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      value: "",
+      handleSubmit: token => props.handleSubmit(token, this.handleResponse)
+    };
+
+    this.handleChange = this.handleChange.bind(this);
+    this.handleResponse = this.handleResponse.bind(this);
+  }
+
+  handleChange(event) {
+    this.setState({ value: event.target.value });
+  }
+
+  /**
+   * In case of a incorrect token, update state to display error. In case of a correct
+   * token, update state to render null.
+   */
+  handleResponse(message) {
+    window.alert(message);
+  }
+
+  render() {
+    const token = this.state.value;
+    return (
+      <form onSubmit={() => this.state.handleSubmit(token)}>
+        <label>
+          Name:
+          <input
+            type="text"
+            value={this.state.value}
+            onChange={this.handleChange}
```

Comment:
If you change this to () => this.handleChange(), you don't need to bind in the constructor.